kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/internal/asset/provider/AssetOptions.java

```diff
@@ -103,11 +103,8 @@ private void addChannel(final Channel channel) {
      *
      * @param properties
      *            the properties to retrieve the channels from
-     * @throws NullPointerException
-     *             if the argument is null
      */
     private void checkChannelAvailability(final Map<String, Object> properties) {
-        requireNonNull(properties, message.propertiesNonNull());
```

Comment:
The contract of this method is to use a map which is not null. Even though the map to be passed to this method in this class is not null, the contract should apply such preconditions.